src/core/notifier/handler/notification/scan_image_handler.go

```diff
@@ -0,0 +1,127 @@
+package notification
+
+import (
+	"errors"
+	"fmt"
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/core/notifier/model"
+	"github.com/goharbor/harbor/src/pkg/notification"
+	"strings"
+)
+
+// ScanImagePreprocessHandler preprocess chart event data
+type ScanImagePreprocessHandler struct {
+}
+
+// Handle preprocess chart event data and then publish hook event
+func (si *ScanImagePreprocessHandler) Handle(value interface{}) error {
+	// if global notification configured disabled, return directly
+	if !config.NotificationEnable() {
+		log.Debug("notification feature is not enabled")
+		return nil
+	}
+
```

Comment:
if !config.WithClair(){
  return nil
}

Reply:
the logic will go through this judge statement before triggering this event.